tests/subsys/net/lib/nrf_cloud/nrf_cloud_test/src/nrf_cloud_test.c

```diff
@@ -0,0 +1,609 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdio.h>
+#include <net/nrf_cloud.h>
+#include <nrf_cloud_transport.h>
+#include <nrf_cloud_fsm.h>
+#include <nrf_cloud_codec.h>
+#include <zephyr/fff.h>
+
+DEFINE_FFF_GLOBALS;
+
+/* Fake functions declaration */
+FAKE_VALUE_FUNC(int, nct_init, const char *);
```

Comment:
I would recommend moving all of your fakes into an `nrf_cloud_fakes.h` file and then including that.

Reply:
Adding `nrf_cloud_fakes.h` will create multiple definition of the mocked functions when building a test runner (since the test runner will add that file again). Without test runner, tests will need to be added manually in the main function, and a suite tear down function will be needed (test runner have them added/created automatically). I don't see much benefit with having `nrf_cloud_fakes.h`

Reply:
> Adding nrf_cloud_fakes.h will create multiple definition of the mocked functions when building a test runner (since the test runner will add that file again)

I think an include guard would solve this, right?

But even then, why does the test runner need to include `nrf_cloud_fakes.h`? What do you mean by "add that file again"?

> I don't see much benefit with having nrf_cloud_fakes.h

IMO it would make your test suite easier to read. When you have this many mocks or fakes it is not uncommon to see them separated into another file just so that a person doesn't have to spend their time scrolling through them to read the actual tests. But it's your call, ultimately

Reply:
Right, I forgot about those include guard. The test runner is based on my `nrf_cloud_test.c` file, so it will include all the libs that it has. 
I forgot that use mentioned to separating the FMFU tests, that would be helpful in such case. Thanks.